engine/worker/run.go

```diff
@@ -341,21 +340,16 @@ func teardownBuildDirectory(wd string) error {
 	return os.RemoveAll(wd)
 }
 
-func generateWorkingDirectory() (string, error) {
-	size := 16
-	bs := make([]byte, size)
-	if _, err := rand.Read(bs); err != nil {
-		return "", err
-	}
-	str := hex.EncodeToString(bs)
-	token := []byte(str)[0:size]
-
-	return string(token), nil
-}
+func workingDirectory(basedir string, jobInfo *sdk.WorkflowNodeJobRunData, suffixes ...string) string {
+	var encodedName = base64.StdEncoding.EncodeToString([]byte(jobInfo.NodeJobRun.Job.Job.Action.Name))
```

Comment:
With the local hatchery case, two or more worker will maybe share the same workspace.